Source/Moya+Internal.swift

```diff
@@ -12,25 +12,32 @@ public extension MoyaProvider {
         let stubBehavior = self.stubClosure(target)
         let cancellableToken = CancellableWrapper()
 
+        // Allow plugins to modify response
+        let pluginsWithCompletion: Moya.Completion = { result in
+            var result = result
+            self.plugins.forEach { result = $0.processResponse(result, target: target) }
```

Comment:
I actually like `reduce` better because you can maintain immutability and also it's very clear that the `result` or `request` is being modified.

It might be slightly bulkier, but TBH `requestNormal` is pretty overweight as is 🙈, and I'd be more concerned about refactoring that to make it more readable and highlight the request/response lifecycle better. I'd at least write it like this:
```swift
let pluginsWithCompletion: Moya.Completion = {
  result in
  let processedResult = self.plugins.reduce(result) { $1.processResponse($0, target: target) }
  completion(processedResult)
}
```